File: wikitext-helpers/lyrics_stanza.py
import fileinput
import re

print("; Release Date: xxxx-xx-xx\n\n== Lyrics ==");

def out_buffer(b):
  if len(b) > 0: print('<br/>\n'.join(b),end='\n\n')
  b.clear()

heading = re.compile(r'^  \[ (.*?) \]  $', re.X)
title_prefix = re.compile(r'^  \d+  \.  \s+', re.X)

buffer = []
for line in fileinput.input(encoding="utf-8"):
  line = heading.sub(r"[''\1'']",line.strip())
  if len(buffer) == 0 and (prefix := title_prefix.match(line)):
    print(f"=== {line[prefix.end():]} ===")
    continue
  buffer.append(line) if len(line) > 0 else out_buffer(buffer)

out_buffer(buffer)
print("\n[[Category:Albums (Extreme Metal)]]")

File: wikitext-helpers/lyrics_stanza.rb
@buffer = []

def out_buffer()
  puts @buffer.join("<br/>\n"), "" unless @buffer.empty?; 
  @buffer.clear
end

puts "; Release Date: xxxx-xx-xx\n\n== Lyrics ==";
STDIN.each_line do |l|
  l.strip!.sub!(%r{^\[(.*?)\]$},%q![''\1'']!)        # convert [xxxx] to [''xxx'']
  if @buffer.empty? and l.match(%r{^\d+\.\s+}) then  # is it a song title?
    puts "=== #{$'} ==="
    next
  end 
  if l.empty? then out_buffer() else @buffer << l end
end
out_buffer()
puts "\n[[Category:Albums (Extreme Metal)]]";

# vim: sw=2 expandtab
